engines/resultset.go

```diff
@@ -1,6 +1,16 @@
 package engines
 
-import "io"
+import (
+	"io"
+)
+
+// A ReadSeekCloser wraps some basic io operations such that we can do
+// retries in artifact upload logic
+type ReadSeekCloser interface {
```

Comment:
Since this is an exported interface, doesn't the interface already document what it implements?  

I'm confused by "This interface is used to expose file objects from containers." since "containers" is a very linux container thing and this same interface would be used when extracting files from a windows environment, right? Or from a file created on a host by a linux/os x process?